libcxx/test/libcxx/algorithms/vectorization.compile.pass.cpp

```diff
@@ -0,0 +1,19 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// We don't know how to vectorize algorithms on GCC
+// XFAIL: gcc
+
+// This test ensures that we enable the vectorization of algorithms on the expected
+// platforms.
+
+#include <algorithm>
+
+#if !_LIBCPP_VECTORIZE_ALGORITHMS
```

Comment:
This seems slightly better.
```suggestion
#if !defined(_LIBCPP_VECTORIZE_ALGORITHMS) || !_LIBCPP_VECTORIZE_ALGORITHMS
```

Reply:
Since we check for the positive, the test would fail if `_LIBCPP_VECTORIZE_ALGORITHMS` were not defined. I'm not certain what the additional test buys us? I guess I can test the test by adding an `#error` if the macro isn't defined.

Reply:
Do we have other tests the ensure the macro is always defined? If yes then I'm fine omitting that test here.
Maybe if there is no such test, we should do that in a separate test.

Reply:
I think we use `-Wundef` to catch that very issue. If we do `#if ! _LIBCPP_VECTORIZE_ALGORITHMS` and `_LIBCPP_VECTORIZE_ALGORITHMS` isn't defined, I'd expect a compiler diagnostic.

That being said, I've made the change you suggested to check for `#ifndef _LIBCPP_VECTORIZE_ALGORITHMS` explicitly and say that the test is broken in that case, which was your original intent IIUC.